src/order/complete_lattice.lean

```diff
@@ -490,10 +490,9 @@ by { convert h1.supr_comp g, exact (funext h2).symm }
 @[congr] theorem supr_congr_Prop {α : Type*} [has_Sup α] {p q : Prop} {f₁ : p → α} {f₂ : q → α}
   (pq : p ↔ q) (f : ∀x, f₁ (pq.mpr x) = f₂ x) : supr f₁ = supr f₂ :=
 begin
-  have : f₁ ∘ pq.mpr = f₂ := funext f,
-  rw [← this],
-  refine (function.surjective.supr_comp (λ h, ⟨pq.1 h, _⟩) f₁).symm,
-  refl
+  have := propext pq, subst this,
+  congr' with x,
+  apply f
```

Comment:
This is just incidental to the PR, right? (I'm not objecting.)

Reply:
Yes, I was writing `finprod_congr_Prop`, and noticed that this lemma can be proved in the same way.